src/components/HybridAppMiddleware/index.tsx

```diff
@@ -49,6 +50,20 @@ function HybridAppMiddleware({children, authenticated}: HybridAppMiddlewareProps
     const [sessionEmail] = useOnyx(ONYXKEYS.SESSION, {selector: (session) => session?.email});
     const [completedHybridAppOnboarding] = useOnyx(ONYXKEYS.NVP_TRYNEWDOT, {selector: onboardingStatusSelector});
 
+    const maxTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+    // We need to ensure that the BootSplash is always hidden after a certain period.
+    useEffect(() => {
+        if (!NativeModules.HybridAppModule) {
+            return;
+        }
+
+        maxTimeoutRef.current = setTimeout(() => {
+            Log.info('[HybridApp] Forcing transition due to unknown problem', true);
+            setStartedTransition(true);
+            setExitTo(ROUTES.HOME);
+        }, 3000);
```

Comment:
I guess we can look into the underlying problem later. Is 3 seconds good enough to reliably solve this problem though?

Reply:
It's just a security measure that @mateuuszzzzz already wanted to introduce for some edge cases. The worst thing that will happen is that a user will see some transitioning between screens which we wanted to cover, though 😄

Reply:
Hmm yes I think we only want to [add a `setTimeout` as a last resort](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md?plain=1#L56), is there nothing else we can do here?

Reply:
I was thinking about it, but I wanted to keep it to make sure we never get a "bootsplash got frozen" situation. I tried to play with state, but I think this is the safest from the UX perspecitve anyways. Also, it won't any additional re-renders/state changes.

I'm not 100% happy with this solution either, but for now I think we can stay with it. When we have some bandwidth in the future I think we can return to this issue to solve it in a cleaner way.